### ISSUE TITLE
Block/model spawning issues

### ISSUE THREAD

**Issue by TheDuckCow** [Moo-Ack-Productions/MCprep#267]
With v3.3.0 just released, there are a few known issues already with the model spawner. Rather than wait until every single obscure block is functioning, the update instead comes a bit sooner but with some known issues. There may be more blocks than these that don't immediately work. *There are likely more than the below, these are just the known ones so far*

**Blocks that spawn apparently without any geometry**
  - beds
  - banner
  - barrel
  - shulker box
  - moving piston
  - chest
  - ender chest

**Blocks that spawn with at least some geo, but have other artifacts:**
  - Bell between walls (just the bar, no bell)
  - Cake kiinda works.. but weird, 'exploded' view
  - Brewing stand technically works.. it's just split into multiple pieces :/
  - Enchanting table (does spawn, but obv. no book)
  - Hopper: spawns, but something seems off about textures.
  - Cauldron (similar issue to hopper on top it seems)
  - Scaffolding (similar issue to hopper, but the "legs")
  - Sunflower just has z-fighting, likely true for any flower style model. 
  - End_portal
  - Composter has messed up UVs

**Blocks that result in errors on spawn:**
 - cube_all (likely should be removed anyways, as a meta-constructor)

**Materials could be auto-animated where relevant:**
 - Seagrass (animate textures doesn't work after the fact evidently)

Because the generation is entirely procedural from json files, these may take some time to fix

**Comment by Zachadactyl** [comment]
Yes, the cauldron definitely has issues, I use BSS Cycles Edit for the cauldron instead.  I tried UV mapping but got lost whoops.

**Comment by TheDuckCow** [comment]
There are some major improvements needed for this operator. I was tempted to move the feature into an experimental flag, but since it's been out there long enough and enough people are using it, I'm going to keep it as is. Some extra try catching has been added, but major issues like UV issues still remain. I'm going to move this bug report into a later milestone, but noting a large number of reported bugs will now be resolved thanks to the extra try/catch and list check before referencing code added previously.

I'm also hoping to add a way to capture what kinds of blocks people are attempting to generate to also get a sense where issues occur, possibly a mini survey asking if the block was generated correctly or not. I will not hold up releases on this however, and would prefer to hear from users on this thread (yes, anyone can reply!) with what they tried to spawn, attaching json files if appropriate (ie if not coming from the MCprep built in ones)

**Comment by TheDuckCow** [comment]
Side note @StandingPadAnimations, idk if you have any interest in taking a crack at this or debugging some of the example failure cases here, I keep putting this off focussing on other items. No worries if not interested.

**Comment by MEESTERSoupCan** [comment]
Another object that can't load is the Held model of the Trident.
[Trident.zip](https://github.com/TheDuckCow/MCprep/files/9050039/Trident.zip)
![image](https://user-images.githubusercontent.com/34176597/177427486-8576043c-c642-411c-8311-d6215337998a.png)

(I was told to post these here in a support email)

**Comment by zNightlord** [comment]
[trident.txt](https://github.com/TheDuckCow/MCprep/files/9051832/trident.txt)
You might need to change the .txt file to json and import the texture manually.

Unfortunately all entities (mob/tile) they all hardcoded in Java in their model or renderer
https://minecraft.fandom.com/wiki/Block_entity#List_of_block_entities

There are problems on how to handle store the hardcoded geometry and bring it. 
Like a blend file ( in Entity Spawner) 
because they are Block Entity and you can bring in the rigged version of it which is more simpler.

or keep on using json ( in Block (model) spawner )
That require a folder to store all the jsons have elements
like this trident file above which I modeled based on how they defined the uv coord and cube origin and dimension. 

But there is a problem for the json method is how to handle variant special case like the bed or chest. 
Bed color variant they don't really store the json file the color version of it in the model folder but in the blockstates instead that also for trapped chest, double and you can't really get the christmas version of it either.

Maybe some of the json model can point to the Entity Spawner to spawn it in or display a prompt for choosing the variants.

**Comment by Tiefseetauchner** [comment]
Wanna add to "Blocks that spawn apparently without any geometry", Water also spawns with 8 unconnected vertices bunched up

**Comment by Sirocco745** [comment]
A note on the End Portal Frame: its geometry generates correctly, but the textures are mixed up. This can be fixed manually after spawning the block by simply reassigning the materials to the correct faces. This goes for both the End Portal Frame with and without an Eye of Ender inside of it. 
The same issue applies to the cake, although it is a slight bit more complicated. The same material bug described with the End Portal Frame occurs, but the geometry of the cake also doesn't generate correctly, requiring manual fixing inside of the .blend file. The solution for the textures is the same as above; reassign materials to their correct faces. The geometry is simply fixed with a couple of Loop Cuts and moving the vertices.
You guys probably already know about these issues specifically, but I thought I'd just add my experiences, as limited as they may be.

**Comment by zNightlord** [comment]
Since the PR for this was merged, the priority for this is lower now *(although there are blocks that haven't added to that ignore list yet)*. 
Metioning #117 because these blocks should considered as meshswap blocks instead.

**Comment by Pro2Vatan** [comment]
block air?
![Снимок экрана 2024-05-02 190525](https://github.com/Moo-Ack-Productions/MCprep/assets/104592160/1336143e-2968-4885-b2ce-1f67043dd6da)

**Comment by TheDuckCow** [comment]
User @Unrevealing added another example for the composter UV issues:

![Image](https://github.com/user-attachments/assets/84efa2de-ba07-4878-bc31-5feea6d0fb9b)

**Comment by TheDuckCow** [comment]
I'm going to mark this as done in the meantime, and call out the remaining issues more as feature improvements.
